src/Npgsql/TypeMapping/GlobalTypeMapper.cs

```diff
@@ -502,7 +502,8 @@ void SetupNetworkHandlers()
                 };
 
                 // Support ReadOnlyIPAddress, which was added to .NET Core 3.0 as an internal subclass of IPAddress
-                if (typeof(IPAddress).GetNestedType("ReadOnlyIPAddress", BindingFlags.NonPublic) is Type readOnlyIpType)
+                var readOnlyIpType = IPAddress.Loopback.GetType();
```

Comment:
Not sure about that one chief - while I am 100% sure that's going to work, depending on that `IPAddress.Loopback` is always going to be a different type doesn't sound that great.

Reply:
The idea is only to not add it if it *isn't* a different type (i.e. in old versions) - so that we don't map IPAddress twice. Do you see a problem here?

Reply:
Only if for some inexplicable reason `IPAddress.Loopback` changes it's type to `IPAddress` and someone attempts to pass a `ReadOnlyIPAddress` as a param. But thinking more about it, it does sound very far fetched.

Reply:
Note that ReadOnlyIPAddress is an internal type... In general, we have no real commitment to supporting it - only to supporting whatever IPAddress.Loopback (and friends) are